src/popper/runner_kubernetes.py

```diff
@@ -110,10 +110,19 @@ def _init_pod_schedule(self):
             pod_host_node = self._config.resman_opts.pod_host_node
 
         elif not self._config.resman_opts.get("persistent_volume_name", None):
-            nodes = [
-                node.metadata.labels["kubernetes.io/hostname"]
-                for node in self._kclient.list_node().items
-            ]
+            nodes = []
+
+            for node in self._kclient.list_node().items:
+                node_role = ""
+                if node.spec.taints.key is not None and len(node.spec.taints) > 0:
+                    node_role = node.spec.taints[0]
+                if node_role != "node-role.kubernetes.io/master" and node_role != "node-role.kubernetes.io/unreachable":
+                    nodes.insert(0, node.metadata.labels["kubernetes.io/hostname"])
+                else if node_role != "node-role.kubernetes.io/unreachable":
```

Comment:
this can be just a simple `else`, we do not need to check the role again here. we are already checking for it in the if part. Also `else if` is expressed as `elif` in python language.

Reply:
You can just add a commit. no need to squash.

Reply:
Ah. I must have misinterpreted you in the past. I thought we were attempting to totally exclude unreachable nodes.